### ISSUE TITLE
The Windows version hangs quite often

### ISSUE THREAD

**Issue by wyf88** [typora/typora-issues#26]
I'm trying Typora 0.7.6 on Windows 10 x64. The program hangs just after I open some certain files, e.g. when I open the "Markdown reference" from the Help menu, after 2 seconds Typora is not responding. In addition, one of the background processes of Typora will use up a single CPU core. 

Another thing is, when this bug occurs, if I just kill Typora in Task Manager, that background process is not closed. Once after I tried 8 times on that problematic file without being aware of this issue, my fan began to cry loudly, and later I found that all of the CPU cores were eaten up. (This does not happen when Typora is closed properly when it does not hang.)

I'll appreciate it if you can take a look at this issue. 

**Update**: As of Version 0.7.7, this problem is not solved.

**Comment by abnerlee** [comment]
I will try to reproduce this on my local PC.

BTW, what's the name of the background process that ate your CPU?
And which windows style did you use in `Preference`? "system native" or "unibody"

**Comment by wyf88** [comment]
The process name is Typora.exe. I was using "system native", but I just tried "unibody" and it hangs as well for the "Markdown reference" file.

**Comment by abnerlee** [comment]
Thanks for the information, 

could you:
1. open them folder from typora->preference->open theme folder, then open its parent folder.
2. open and crash typora like you mentioned, then attach the `typora.log`(in previous folder) to me. Or send typora.log to hi@typora.io if you worry about the privacy.

After that, would you try delete the `C:\Users\{your user name}\AppData\Roaming\Typora` folder and restart typora. see if it will solve the problem or not. If not work, try uninstall, reinstall typora...

Meanwhile, I will look into this issue, but I cannot reproduce this bug now~

**Comment by wyf88** [comment]
1. Sure. The `typora.log` is as follows:
```
{ width: 800,
  height: 750,
  frame: true,
  radii: [ 0, 0, 0, 0 ],
  DirectWrite: true,
  disableAutoHideCursor: false,
  webPreferences: { allowDisplayingInsecureContent: true, zoomFactor: 1 } }
open file: undefined
{ width: 800,
  height: 750,
  frame: true,
  radii: [ 0, 0, 0, 0 ],
  DirectWrite: true,
  disableAutoHideCursor: false,
  webPreferences: { allowDisplayingInsecureContent: true, zoomFactor: 1 } }
open file: C:\Program Files (x86)\Typora\resources\app/Docs/Markdown Reference.md
addRecentDocument: C:\Program Files (x86)\Typora\resources\app/Docs/Markdown Reference.md
```
It does not seem to contain suspicious information. 

2. I've tried both delete that folder -> restart Typora and uninstall Typora -> delete that folder -> reinstall Typora. The problem still occurs. So maybe it's a problem to Windows 10? I may try Typora on different versions of Windows when I have time, and report my result to you.

**Comment by wyf88** [comment]
Well, I tried on another PC with Windows 7 x64, and a virtual machine with Windows XP. 
On Windows 7, Typora hangs as well. Here is the log:
```
{ width: 683,
  height: 576,
  frame: true,
  radii: [ 0, 0, 0, 0 ],
  DirectWrite: true,
  disableAutoHideCursor: false,
  webPreferences: { allowDisplayingInsecureContent: true, zoomFactor: undefined } }
open file: undefined
{ width: 683,
  height: 576,
  frame: true,
  radii: [ 0, 0, 0, 0 ],
  DirectWrite: true,
  disableAutoHideCursor: false,
  webPreferences: { allowDisplayingInsecureContent: true, zoomFactor: 1.25 } }
open file: C:\Program Files (x86)\Typora\resources\app/Docs/Markdown Reference.md
addRecentDocument: C:\Program Files (x86)\Typora\resources\app/Docs/Markdown Reference.md
```
On Windows XP, Typora won't run.

**Comment by abnerlee** [comment]
Thanks for the testing. 

I have received one similar bug reports that "Rendering with Latex will cause 100% CPU running", so now I'm wondering if they are the same bug.

Could you try open a md file with content:

```
$$
\mathbf{V}_1 \times \mathbf{V}_2 =  \begin{vmatrix} 
\mathbf{i} & \mathbf{j} & \mathbf{k} \\
\frac{\partial X}{\partial u} &  \frac{\partial Y}{\partial u} & 0 \\
\frac{\partial X}{\partial v} &  \frac{\partial Y}{\partial v} & 0 \\
\end{vmatrix}
$$
```

or just copy and paste it to typora with ctrl+shift+v and see if typora will crash or not?

**Comment by wyf88** [comment]
Yes, Typora hangs only 1 second after I paste the content into the window. 

There's another weird thing. After that, Typora refuses to render any LaTeX code at all. Last night I could open my own manuscript with lots of maths without a problem. But now it only shows as LaTeX code. If I create a new document and input some LaTeX it won't render either.

**Comment by be5invis** [comment]
@abnerlee Perhaps I think you can replace Mathjax with KaTeX, the latter one is must faster and reliable.

**Comment by abnerlee** [comment]
@be5invis Thanks for the advice. 

But according to <http://www.intmath.com/cg5/katex-mathjax-comparison.php>, MathJax now still covers more LaTeX syntax. And because Typora's always use cache + partial update logic, and MathJax would do rendering asynchronously, it seems we can cover the performance gap.

The issue that typora hangs on math expressions and #20 , is more like a bug rather than performance issue.

I will look into this, or update to latest MathJax lib. If still unlucky, then I'll consider switching to KaTeX.

**Comment by DanBennettUK** [comment]
Just to add to this hanging issue... While I'm typing, it takes a second or two for text to appear in Typora.

And on that, sometimes the cursor goes back to the start of the word/sentence whilst I'm typing. So I end up typing something like "easoRns" instead of "Reasons" o_O

**Comment by abnerlee** [comment]
The LaTeX rendering and Maximum call stack size exceeded should have being fixed in 0.8.3,

@DanBennettUK, I think your bug relates to #68

**Comment by Lovfer** [comment]
Can you porvide a xp version of Typera?
